llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -24405,6 +24405,22 @@ bool RISCVTargetLowering::isLegalStridedLoadStore(EVT DataType,
   return true;
 }
 
+bool RISCVTargetLowering::isLegalSpeculativeLoad(EVT DataType,
+                                                 Align Alignment) const {
```

Comment:
Nit, I guess this is checking for the legality of `vp.load.ff`, should this be renamed `isLegalFaultOnlyFirstLoad` to be consistent? 

I think it might also avoid confusion with regular wide loads that happen to be speculative because of the `dereferenceable` attribute